scalyr_agent/agent_main.py

```diff
@@ -739,12 +751,14 @@ def __detailed_status(self, data_directory, status_format="text"):
             )
             return 1
 
+        return_code = 0
         fp = open(status_file)
         for line in fp:
             print(line.rstrip())
+            if health_check and line.rstrip() != "Health check: Good":
```

Comment:
I think we should, even if we are doing `status -v` (so not an explicit health check), we should return a non-zero status code if the health check has failed.  We should also return different return code, like 2.  That way, scripts can tell the difference between the agent not running and a failed health code.

Of course, if we do `status -v` and the agent does not have health checks enabled, then we shouldn't return a status code based on the health check.    We could do that by only emitting the health check information in the `status -v` output if the health check is enabled.  Then, in this code, we just look for a line that starts with `Health check` and see if it is good or bad... setting some variable to True or False depending on the status.  If we don't find any line that starts with `Health check` then we set that variable to None.

We should also be sure this handles the JSON format for the status correctly as well.